ghc-android/src/user-scripts/build-gmp.sh

```diff
@@ -4,7 +4,7 @@
 ####################################################################################################
 
 echo Downloading gmp "$GMP_RELEASE"
-curl -o "$TARDIR/$GMP_TAR_FILE" https://gmplib.org/download/gmp/"$GMP_TAR_FILE" 2>&1
+curl -o "$TARDIR/$GMP_TAR_FILE" https://ftp.gnu.org/gnu/gmp/"$GMP_TAR_FILE" 2>&1
 check_md5 "$GMP_TAR_PATH" "$GMP_MD5"
```

Comment:
While not in the scope of this PR, I noticed that the script checks MD5 of the downloaded file. Perhaps it makes sense to switch to a not yet vulnerable secure hash function in some future patch-set?